installer/configurator/telemetry_sender.go

```diff
@@ -0,0 +1,278 @@
+)
+
+type TelemetrySender struct {
+	AgentEndpoint    string
+	RuntimeID        string
+	Application      Application
+	Host             Host
+	client           *http.Client
+	seqID            int
+	telemetryEnabled bool
+	metricsEnabled   bool
+	logsEnabled      bool
+}
+
+type Application struct {
+	ServiceName     string `json:"service_name"`
+	Env             string `json:"env"`
+	ServiceVersion  string `json:"service_version"`
+	TracerVersion   string `json:"tracer_version"`
+	LanguageName    string `json:"language_name"`
+	LanguageVersion string `json:"language_version"`
+}
+
+type Host struct {
+	Hostname      string `json:"hostname"`
+	OS            string `json:"os"`
+	OSVersion     string `json:"os_version"`
+	Architecture  string `json:"architecture"`
+	KernelName    string `json:"kernel_name"`
+	KernelRelease string `json:"kernel_release"`
+	KernelVersion string `json:"kernel_version"`
+}
+
+type Telemetry struct {
+	APIVersion  string      `json:"api_version"`
+	RequestType string      `json:"request_type"`
+	TracerTime  int64       `json:"tracer_time"`
+	RuntimeID   string      `json:"runtime_id"`
+	SeqID       int         `json:"seq_id"`
+	Application Application `json:"application"`
+	Host        Host        `json:"host"`
+	Payload     interface{} `json:"payload"`
+}
+
+type MetricData struct {
+	Metric string    `json:"metric"`
+	Points [][]int64 `json:"points"`
+	Type   string    `json:"type"`
+	Tags   []string  `json:"tags,omitempty"`
+}
+
+type GenerateMetricsPayload struct {
+	Namespace string       `json:"namespace"`
+	Series    []MetricData `json:"series"`
+}
+
+type LogMessage struct {
+	Message string `json:"message"`
+	Level   string `json:"level"`
+	Count   int    `json:"count,omitempty"`
+	Tags    string `json:"tags,omitempty"`
+}
+
+type LogsPayload struct {
+	Logs []LogMessage `json:"logs"`
+}
+
+func NewTelemetrySender(agentHost, serviceName, env string) (*TelemetrySender, error) {
+	app, err := initApplication(serviceName, env)
+	if err != nil {
+		return nil, NewInstallerError(TelemetryError, fmt.Errorf("failed to initialize application info: %w", err))
+	}
+
+	host := initHost()
+
+	return &TelemetrySender{
+		AgentEndpoint:    agentHost + AgentEndpoint,
+		RuntimeID:        uuid.New().String(),
+		Application:      app,
+		Host:             host,
+		client:           &http.Client{},
+		seqID:            0,
+		telemetryEnabled: envBool("DD_INSTRUMENTATION_TELEMETRY_ENABLED"),
+		metricsEnabled:   envBool("DD_TELEMETRY_METRICS_ENABLED"),
+		logsEnabled:      envBool("DD_TELEMETRY_LOG_COLLECTION_ENABLED"),
+	}, nil
+}
+
+func initApplication(serviceName, env string) (Application, error) {
+	if serviceName == "" || env == "" {
+		return Application{}, NewInstallerError(TelemetryError, fmt.Errorf("serviceName and env are required"))
+	}
+
+	return Application{
+		ServiceName:     serviceName,
+		Env:             env,
+		ServiceVersion:  InstallerVersion,
+		TracerVersion:   InstallerVersion,
```

Comment:
[recommendation] If we just use the installer version, I am afraid it will be confusing for one to know if it's actually a tracer version or something else. How about we prefix this with "nginx-installer-"?

Reply:
It's already part of the service name, so it wouldn't be adding any extra information